manifests/qdevice.pp

```diff
@@ -0,0 +1,107 @@
+# This class performs the configuration of the qdevice daemon on a target node.
+# Note that this requires corosync 2.x and must never be deployed on a node
+# which is actually part of a cluster. Additionally, you will need to open the
+# correct firewall ports for both pcs, and the actual quorum device as shown in
+# the included example.
+#
+# @param sensitive_hacluster_hash [Sensitive[String]]
+#   The password hash for the hacluster user on this quorum device node. This 
+#   is currently a mandatory parameter because pcsd must be used to perform the
+#   quorum node configuration. 
+#
+# @param package_pcs [String]
+#   Name of the PCS package on this system. This is specified via hiera by
+#   default as 'pcs'.
+#
+# @param package_corosync_qnetd [String]
+#   Name of the corosync qnetd package for this system. The default value of
+#   'corosync-qnetd' will be provided via hiera if left unspecified.
+#
+# @summary Performs basic initial configuration of the qdevice daemon on a node.
+#
+# @example Quorum node with default password & configuring the firewall
+#   include firewalld
+#
+#   class { 'corosync::qdevice':
+#     sensitive_hacluster_hash => $sensitive_hacluster_hash,
+#   }
+#   contain 'corosync::qdevice'
+# 
+#   # Open the corosync-qnetd port
+#   firewalld::custom_service { 'corosync-qdevice-net':
+#     description => 'Corosync Quorum Net Device Port',
+#     port        => [
+#       {
+#         port     => '5403',
+#         protocol => 'tcp',
+#       },
+#     ],
+#   }
+#   firewalld_service { 'corosync-qdevice-net':
+#     ensure  => 'present',
+#     service => 'corosync-qdevice-net',
+#     zone    => 'public',
+#   }
+#
+#   # Configure general PCS firewall rules
+#   firewalld_service { 'high-availability':
+#     ensure  => 'present',
+#     service => 'high-availability',
+#     zone    => 'public',
+#   }
+#
+# @see https://www.systutorials.com/docs/linux/man/8-corosync-qnetd/
+class corosync::qdevice (
+  String $package_pcs                         = undef,
```

Comment:
I think the combination of `String` and `undef` isn't valid. My preference would be to allow undef and default to `Optional[String[1]]` as datatype

Reply:
I was trying to ensure that the $package_pcs parameter was mandatory for the qdevice parameter. The idea being that for RedHat family operating systems the default is sourced from hiera by default and any other OS would require explicitly setting the parameter. Is there another way you typically achieve this functionality?

Reply:
If the parameter should be mandatory, it should be:
```
String[1] $package_pcs,
```

I'm fine with that as well.

Reply:
Perfect. I've updated to match.